nemo/collections/asr/models/label_models.py

```diff
@@ -466,25 +468,74 @@ def infer_file(self, path2audio_file):
         del audio_signal, audio_signal_len
         return emb, logits
 
-    def get_label(self, path2audio_file):
+    @torch.no_grad()
+    def infer_segment(self, segment):
+        """
+        Args:
+            segment: segment of audio file
+
+        Returns:
+            emb: speaker embeddings (Audio representations)
+            logits: logits corresponding of final layer
+        """
+        segment_length = segment.shape[0]
+
+        device = self.device
+        audio = np.array([segment])
+        audio_signal, audio_signal_len = (
+            torch.tensor(audio, device=device),
+            torch.tensor([segment_length], device=device),
+        )
+        mode = self.training
+        self.freeze()
+
+        logits, emb = self.forward(input_signal=audio_signal, input_signal_length=audio_signal_len)
+
+        self.train(mode=mode)
+        if mode is True:
+            self.unfreeze()
+        del audio_signal, audio_signal_len
+        return emb, logits
+
+    def get_label(self, path2audio_file: str, segment_duration: float = np.inf, num_segments: int = 1):
         """
         Returns label of path2audio_file from classes the model was trained on.
         Args:
-            path2audio_file: path to audio wav file
+            path2audio_file (str): path to audio wav file
+            segment_duration (float): random sample duration in seconds
+            num_segments (int): number of segments of file to use for majority vote
 
         Returns:
             label: label corresponding to the trained model
         """
-        _, logits = self.infer_file(path2audio_file=path2audio_file)
+        audio, sr = librosa.load(path2audio_file, sr=None)
```

Comment:
can you replace this with `sf.read` it is much faster

Reply:
done

Reply:
I know it might not have been originally designed to support reading mp3 or multi-channel (stereo) wav files, but it was able to do so in the past with librosa.load, however, it may result in errors after switching to sf.read. Should we consider adding support for more formats or stick to using librosa.load for consistency?